source/1_engine/rex_engine/src/filesystem/vfs.cpp

```diff
@@ -311,6 +286,42 @@ namespace rex
           });
     }
 
+    void init(rsl::string_view root)
+    {
+      if(rsl::optional<rsl::string_view> arg = cmdline::get_argument("Root"))
+      {
+        root = arg.value();
+      }
+
+      if(root.empty())
+      {
+        WIN_CALL(GetCurrentDirectoryA(g_root.max_size(), g_root.data()));
+        g_root.reset_null_termination_offset();
+      }
+      else if(is_abs(root))
+      {
+        g_root = root;
+      }
+      else
+      {
+        rsl::medium_stack_string cwd;
+        WIN_CALL(GetCurrentDirectoryA(cwd.max_size(), cwd.data()));
+        cwd.reset_null_termination_offset();
+
+        g_root = cwd;
+        g_root += "/";
+        g_root += root;
+      }
+
+      g_is_initialized  = true;
+      g_keep_processing = true;
```

Comment:
Maybe bit overkill but we have a statecontroller we could utilize it here.

Reply:
these are just flags, the state controller is more to have multiple states at once (and used with an enum) so I think it's a bit overkill to use it here